### PR TITLE
Split string with lat and lon and sparse them.

### PR DESCRIPTION
**Pull request by AlbanSagouis**
## Description
`pz_split_llstr_string` splits one string where latitude and longitude are separated by , ; . or a space character.
`pz_split_llstr` takes a vector of strings and calls `pz_split_llstr_string` to split them.

R function `parse_llstr` (that existed already) calls the c++ function `pz_split_llstr`, applies parsing to latitude and longitude values with `parse_lat` and `parse_lon`, and returns a `data.frame` with lat and lon in decimal degrees.

## Related Issue
This fix addresses issue #3 

## Usage
```
parse_llstr(c("N4'55'33', E112'13'40'",
                    "N4'55'33'; E112'13'40'",
                    "N4'55'33' E112'13'40'"))

       lat      lon
1 4.925833 112.2278
2 4.925833 112.2278
3 4.925833 112.2278
```

### CODE REVIEW

**Comment by sckott** [comment]
thanks @AlbanSagouis - having a look

**Comment by sckott** [comment]
For pull requests, I prefer if they are concentrated on a single topic. scrub is a separate issue, and I think requires a some discussion. Can you remove the scrub related changes, and put on a separate branch and PR?

File: src/desktop.ini [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[ViewState]
```
Comment (sckott) [inline src/desktop.ini:1]: please add `src/desktop.ini` to the .Rbuildignore and .gitignore files

File: R/parse_llstr.R [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#' parse_llstr("N05.03062,E101.75172")
+#' parse_llstr("N4.9196, E101.345")
+#' parse_llstr("N4.9196, E101.346")
+#' parse_llstr("N4.9196, E101.347")
+#' # no comma
+#' parse_llstr("N4.9196 E101.347")
+#' #' # no space
+#' parse_llstr("N4.9196E101.347")
+#'
+#' DMS
```
Comment (sckott) [inline R/parse_llstr.R:20]: examples fail because you need to comment out non-code lines with `#`

File: R/parse_llstr.R [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+#' parse_llstr("N04.82344, E101.61320")
+#' parse_llstr("N 04.25164, E 101.70695")
+#' parse_llstr("N05.03062, E101.75172")
+#' parse_llstr("N05.03062,E101.75172")
+#' parse_llstr("N4.9196, E101.345")
+#' parse_llstr("N4.9196, E101.346")
+#' parse_llstr("N4.9196, E101.347")
+#' # no comma
+#' parse_llstr("N4.9196 E101.347")
+#' #' # no space
```
Comment (sckott) [inline R/parse_llstr.R:17]: just one `#'`

File: R/parse_llstr.R [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+#' parse_llstr("N4.9196E101.347")
+#'
+#' DMS
+#' parse_llstr("N4 51′36″, E101 34′7″")
+#' parse_llstr(c("4 51′36″S, 101 34′7″W", "N4 51′36″, E101 34′7″"))
+#'
+#' str <- "N 04.1683, E 101.5823"
```
Comment (sckott) [inline R/parse_llstr.R:24]: delete these last two lines of examples, looks like `str` isn't used

File: R/parse_llstr.R [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+parse_llstr <- function(str) {
+
+  assert(str, 'character')
+  tmp <- pz_split_llstr(scrub(str))
+
+  return(
+    data.frame(
```
Comment (sckott) [inline R/parse_llstr.R:31]: to match what `parse_lon_lat`, let's do `lon` as the first column and `lat` as second column

Comment (AlbanSagouis) [reply R/parse_llstr.R:31]: Done but isn't it confusing that the string expected in argument is "latitude, longitude" (the most common form) and the result longitude and latitude?

**Comment by sckott** [comment]
Thanks @AlbanSagouis - two items: 

- Please answer my question about boost
- did you address the c++ errors I showed above?

**Comment by AlbanSagouis** [comment]
Hi @sckott ,
My previous commit addresses the issues arising when several spaces were present. Now the function deletes any extra spaces and the c++ line giving the error you showed (see below) was deleted.

```
pz_split_llstr.cpp:19:54: warning: multi-character character constant [-Wmultichar]
    boost::split(splitstr, x, [](char c){return c == ', ';});
```

I'll look into replacing boost functions with std functions. I'm thinking of using `str.find` to look for the separator and `str.substr` to select coordinate elements OR `std::regex_replace`

**Comment by AlbanSagouis** [comment]
Hi Scott,
My last commit made the code not dependent on boost anymore.
I used `std::regex_replace` to alternatively delete latitude or longitude and pick up coordinates separately.

**Comment by sckott** [comment]
thanks, i'll have another look at the code locally

File: R/parse_llstr.R (hunk 5 above)
Comment (sckott) [reply R/parse_llstr.R:30]: You're right, i think it makes sense to give the output columns in order of input. Go ahead and change it back.

**Comment by sckott** [comment]
Thanks for fixing that warning, and removing Boost, that should make installation easier when users are installing development versions 

It looks like `parse_llstr()` is sensitive to order of inputs. Do you think it could work if the user gives longitude first in the input string? For example: `E 101.5823, N 04.1683`

File: R/parse_llstr.R (hunk 5 above)
Comment (AlbanSagouis) [reply R/parse_llstr.R:30]: OK, I reversed it in this commit  
##[https://github.com/ropensci/parzer/pull/24/commits/dbf10de669460955e068de815edb8d63d660ed2b](url)

**Comment by AlbanSagouis** [comment]
> 
> It looks like `parse_llstr()` is sensitive to order of inputs. Do you think it could work if the user gives longitude first in the input string? For example: `E 101.5823, N 04.1683`

Hi,
It is definitely sensitive to the order. If there are letters in the coordinate string, we could add the feature but if not, it gets tricky to make sure latitude is first.

In my [original implementation](https://github.com/AlbanSagouis/iClean/blob/master/R/coordinate_conversions.R), order was checked based on the letters. When no letter was given, there was an argument for that: if `assume_good_order` argument was TRUE, latitude was considered first. When `assume_good_order` was FALSE, the function would return NA whenever there was no letter to properly indicate the order.

We could also add an argument to let the user set the order. ` order = 'latlon' or 'lonlat'`

**Comment by sckott** [comment]
Thanks for the fixes.  

Let's move on now wrt the order of inputs - we can return to that later if there's users that want to be able to control order of lat and lon in strings.

**Pull request merged**